modules/Hoot/managers/layerManager.js

```diff
@@ -287,16 +303,21 @@ export default class Layers {
                 input2Name = tags.input2Name;
 
             if ( (input1 && input1Name) && (input2 && input2Name) ) {
-                let params1 = {
-                    name: input1Name,
-                    id: input1,
-                    color: 'violet'
-                };
+                let loaded = Object.values(Hoot.layers.loadedLayers).find(l => l.id === tags.input1),
```

Comment:
This was where the undefined problem was occurring during the review,

Reply:
Since the params did not include a refType, then this gets passed to loadLayer, which tries to update hash using params.refType, but without checking if params.refType is undefined.